transcom/mymove
# [MB-3389] TIO can authorize payment

## Description

This PR adds the **Complete Request** step to the Payment Request Review page, and implements the "Authorize Payment" handler. If successful, the payment request status should be updated to `Reviewed` and the user should be redirected back to the Payment Request index page.

If the payment request has any service items that have _not_ been approved (have the `Requested` status), the payment request cannot get the `Reviewed` status and an error will be displayed to the user.

Outstanding questions:
- Should `PaymentRequest.ReviewedAt` be updated also?

## Reviewer Notes

There is an outstanding bug that results in:
  - the total approved amount is not updated when you Approve/Reject items until refreshing the page
  - after making a selection for a service item, and then clicking away from and back to that service item, your selection is not shown until refreshing the page.

This PR does not cause or fix this bug.

## Setup

- Make sure your application has data seeded (`make db_dev_e2e_populate`)
- Start the application (`make server_run` `make client_run`)
- Log into the Office app as a TIO user and click on any of the payment requests listed
  - Note the value in the Status column of the Payment Requests table (should be `PENDING`)
- Test the happy path:
  - Choose "Approve" or "Reject" for all of the payment request service items in the sidebar,
  - After the last item, you should be able to click "Authorize Payment" and be redirected back to the Payment Requests table. The status should now say `REVIEWED`
- Test the unhappy path:
  - Click into another pending payment request
  - This time, without selecting "Approve" or "Reject" for every service item, click the Next button until you get to the Complete Request step.
  - Click "Authorize Payment" and instead of changing the status, you should see an error message that says you must approve or reject all service items before reviewing this payment request.

## Code Review Verification Steps

 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/backend#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3389) for this change

## Screenshots

![image](https://user-images.githubusercontent.com/2723066/88851655-ade3e900-d1b2-11ea-9362-f1975af582c0.png)

![image](https://user-images.githubusercontent.com/2723066/88851400-56457d80-d1b2-11ea-854b-b025048da873.png)
